src/supabase/queries/cases.ts

```diff
@@ -37,6 +37,55 @@ export async function getCaseIdsFromUserId(
   }
 }
 
+export async function getCaseById(caseId: CaseUid): Promise<CasePartial> {
+  try {
+    const { data } = await supabase.from('cases').select().eq('caseId', caseId);
+    if (!data) {
+      throw new Error('case not found');
+    }
+    return formatCase(data[0]);
+  } catch (error) {
+    console.warn('(getCaseById)', error);
+    throw error;
+  }
+}
+
+export async function isValidCase(caseId: CaseUid): Promise<boolean> {
+  const { data } = await supabase.from('cases').select().eq('caseId', caseId);
+  if (!data) {
+    return false;
+  }
+  return data.length !== 0;
+}
+
+export async function uploadCase(caseId: CaseUid | undefined) {
```

Comment:
Let's not allow `| undefined` here. Also wrap in a `try-catch` and use `console.warn` instead of `console.log` for consistency.